### DISCUSSION TITLE
How to use react hook form with custom input

### DISCUSSION THREAD

**Discussion by vinodhum** [react-hook-form/react-hook-form#1995] [Q&A] [answered]
Hi, I am trying to make react hook form work with ionic input using custom register, link to stackblitz below.
If I use controller/control dirty field and isValid not update on change and if I use custom register, defaultvalues not setting, touched and dirty fields not updating. any idea how to make all the form state keys updated using custom input component.

[https://stackblitz.com/edit/react-fnwfgx?file=Hello.js](url)

Thank you

**Comment by bluebill1049** [comment, marked as answer]
Follow this video tutorial: https://www.youtube.com/results?search_query=hook+form+mwith+iconic&page=&utm_source=opensearch

**Reply by bradrust** [reply]
Have you seen any tutorials that are using the newer 6.0.2 release which has the Controller as part of that release verses a separate package?

I forked a sandbox and change to this new version (because I can't get it to work locally).  
https://codesandbox.io/s/react-hook-form-ionic-input-components-feopo?file=/src/index.js

It appears that something is missing from my local implementation, and the sandbox, or perhaps there is a defect.
